be2-scala/src/test/scala/ch/epfl/pop/model/objects/HashSuite.scala

```diff
@@ -0,0 +1,57 @@
+package ch.epfl.pop.model.objects
+
+import org.scalatest.{FunSuite, Matchers}
+
+class HashSuite extends FunSuite with Matchers {
+  test("Hash 'fromString' works for a random string") {
+    val hash: Hash = Hash.fromString("PoP")
+    val expected: Hash = Hash(Base64Data("EmE4jL2zWjfJY_BVHYQdX19TtZTQwlqDA--dh_4mo2s="))
+
+    hash should equal (expected)
+  }
+
+  test("Hash 'fromStrings' works against expected fe1-web data") {
+    val hash: Hash = Hash.fromStrings("abcd", "1234")
+    val expected: Hash = Hash(Base64Data("61I7DQkiMtdHFM5VygjbFqrVmn4NAl0wSVxkj6Q5iDw="))
+
+    hash should equal (expected)
+  }
+
+  test("Hash 'fromStrings' works with no string") {
+    val hash: Hash = Hash.fromStrings()
+    val expected: Hash = Hash(Base64Data("47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU="))
+
+    hash should equal (expected)
+  }
+
+  test("Hash 'fromStrings' works with one string") {
+    var hash: Hash = Hash.fromStrings("PoP")
+    var expected: Hash = Hash(Base64Data("fAoSEwSI4qx6brSVryjql0GdkwbMNidcIeJHa8TxGR4="))
```

Comment:
I don't understand why the expected result is not the same as with the first test, since the string is the same. Shouldn't `fromStrings` with one string yield the same as `fromString` ?

Reply:
agreed

Reply:
Mhmmm... I think that's what we are doing in fe1 right ^^' ? `fromString` hashes a string using SHA-256 whereas `fromStrings` hashes using our custom hashing technique (length of the string || string)

Reply:
okay, thanks for the renaming. maybe `fe1` could benefit from the disambiguation too 😅